src/qpx_mmc.cpp

```diff
@@ -0,0 +1,4244 @@
+        get_rw_speeds(drive);
+        drive->parms.read_speed_dvd =
+            (int)(drive->parms.read_speed_kb / drive->parms.speed_mult);
+        if (prev_spd != drive->parms.read_speed_dvd) {
+          //					spd =
+          // drive->parms.read_speed_dvd;
+          spd = max(spd, drive->parms.read_speed_dvd);
+          drive->parms.speed_tbl[idx] = drive->parms.read_speed_dvd;
+          drive->parms.speed_tbl_kb[idx] = drive->parms.read_speed_kb;
+          if (!drive->silent)
+            printf(" RD speed:  %dX (%d kB/s)\n", drive->parms.speed_tbl[idx],
+                   drive->parms.speed_tbl_kb[idx]);
+          prev_spd = drive->parms.read_speed_dvd;
+          idx++;
+          //					if (drive->capabilities &
+          // CAP_REAL_TIME_STREAMING) get_performance(drive);
+        }
+      }
+#ifdef speedidx
+    } else {
+      speedidx = -1;
+#endif
+    }
+  } else {
+    drive->parms.read_speed_kb = -1;
+    if (!set_rw_speeds(drive)) {
+      get_rw_speeds(drive);
+      drive->parms.max_read_speed_kb = drive->parms.read_speed_kb;
+      drive->parms.max_read_speed_cd = (drive->parms.max_read_speed_kb / 176);
+      //			drive->parms.speed_mult=drive->parms.max_read_speed_kb/drive->parms.max_read_speed_cd;
+      drive->parms.speed_mult = 176.4;
+      if (!drive->silent)
+        printf("Maximum CD speed: %dX, %5d kB/s; 1X = %.1f kB/s\nSpeeds:\n",
+               drive->parms.max_read_speed_cd, drive->parms.max_read_speed_kb,
+               drive->parms.speed_mult);
+      idx = 0;
+      prev_spd = 0;
+      for (spd = 1; ((idx < speed_tbl_size) &&
+                     (spd < (drive->parms.max_read_speed_cd + 2)));
+           spd++) {
+        //			for (spd=1; ((idx<speed_tbl_size) && (spd <
+        // 72)); spd++) {
+        drive->parms.read_speed_kb = (int)(spd * (drive->parms.speed_mult + 1));
+        if (!drive->silent)
+          printf("Trying:  %dX (%5d kB/s)\n", spd, drive->parms.read_speed_kb);
+        set_rw_speeds(drive);
+        get_rw_speeds(drive);
+        drive->parms.read_speed_cd =
+            (int)(drive->parms.read_speed_kb / drive->parms.speed_mult);
+        if (prev_spd != drive->parms.read_speed_cd) {
+          //					spd =
+          // drive->parms.read_speed_cd;
+          spd = max(spd, drive->parms.read_speed_cd);
+          drive->parms.speed_tbl[idx] = drive->parms.read_speed_cd;
+          drive->parms.speed_tbl_kb[idx] = drive->parms.read_speed_kb;
+          if (!drive->silent)
+            printf(" RD speed:  %dX (%5d kB/s)\n", drive->parms.speed_tbl[idx],
+                   drive->parms.speed_tbl_kb[idx]);
+          prev_spd = drive->parms.read_speed_cd;
+          idx++;
+          //					if (drive->capabilities &
+          // CAP_REAL_TIME_STREAMING) get_performance(drive);
+        }
+      }
+#ifdef speedidx
+    } else {
+      speedidx = -1;
+#endif
+    }
+  }
+  //	combo_Speed->clear();
+  idx = 0;
+#ifdef speedidx
+  if (!speedidx) {
```

Comment:
Codacy found an issue: [Skipping configuration 'speedidx' since the value of 'speedidx' is unknown. Use -D if you want to check it. You can use -U to skip it explicitly.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)